tarka-ui/src/main/java/com/tarkalabs/tarkaui/components/TUIAppBottomBar.kt

```diff
@@ -0,0 +1,118 @@
+package com.tarkalabs.tarkaui.components
+
+import android.util.Log
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.sizeIn
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.material3.BottomAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.tarkalabs.tarkaui.components.base.IconButtonSize
+import com.tarkalabs.tarkaui.components.base.IconButtonStyle
+import com.tarkalabs.tarkaui.components.base.TUIIconButton
+import com.tarkalabs.tarkaui.components.base.TUIIconButtonTags
+import com.tarkalabs.tarkaui.icons.ArrowRedo24
+import com.tarkalabs.tarkaui.icons.ArrowUndo24
+import com.tarkalabs.tarkaui.icons.Delete24
+import com.tarkalabs.tarkaui.icons.Lasso24
+import com.tarkalabs.tarkaui.icons.SelectObjectSkew24
+import com.tarkalabs.tarkaui.icons.TarkaIcon
+import com.tarkalabs.tarkaui.icons.TarkaIcons
+import com.tarkalabs.tarkaui.theme.TUITheme
+
+
+sealed class BottomBarClickEvent(val index: Int) {
+    class ItemClick(index: Int) : BottomBarClickEvent(index)
+}
+
+
+/**   Displays a customized bottom app bar with listOfIcons,onClickEvent.
+ *
+ * @param modifier The modifier to customize the TUIAppBottomBar container.
+ * @param tags Tags for customizing test accessibility.
+ * @param onClickEvent The callback function to be invoked when the icon is clicked.
+ * @param icons for providing list of icons to be displayed in bottom bar.
+ * @param color: the colors to be applied to the bottom app bar.
+ *
+ * * How to use TUIAppBottomBar()
+ *   TUIAppBottomBar(
+ *     icons = pass the list of icons
+ *     onClickEvent = { event -> Handle event click based on the icon index}
+ *       )
+ */
+@Composable
+fun TUIAppBottomBar(
+    modifier: Modifier = Modifier,
+    icons: List<TarkaIcon?> = emptyList(),
```

Comment:
We should not pass nullable tarkaicon here
```suggestion
    icons: List<TarkaIcon>,
```